controllers/base_controller.js

```diff
@@ -116,54 +132,76 @@ module.exports = function BaseControllerModule(pb) {
+        this.ts.registerLocal('locale', this.ls.language);
+        this.ts.registerLocal('error_success', function(flag, cb) {
             self.displayErrorOrSuccessCallback(flag, cb);
         });
-        this.templateService.registerLocal('page_name', function(flag, cb) {
+        this.ts.registerLocal('page_name', function(flag, cb) {
             cb(null, self.getPageName());
         });
-        this.templateService.registerLocal('localization_script', function(flag, cb) {
+        this.ts.registerLocal('localization_script', function(flag, cb) {
             self.requiresClientLocalizationCallback(flag, cb);
         });
-        this.templateService.registerLocal('analytics', function(flag, cb) {
+        this.ts.registerLocal('analytics', function(flag, cb) {
             pb.AnalyticsManager.onPageRender(self.req, self.session, self.ls, cb);
         });
-        this.templateService.registerLocal('wysiwyg', function(flag, cb) {
+        this.ts.registerLocal('wysiwyg', function(flag, cb) {
             var wysiwygId = util.uniqueId();
 
-            self.templateService.registerLocal('wys_id', wysiwygId);
-            self.templateService.load('admin/elements/wysiwyg', function(err, data) {
+            self.ts.registerLocal('wys_id', wysiwygId);
+            self.ts.load('admin/elements/wysiwyg', function(err, data) {
                 cb(err, new pb.TemplateValue(data, false));
             });
         });
-        this.ts = this.templateService;
-        
+
+
         /**
          *
          * @property activeTheme
          * @type {String}
          */
         this.activeTheme = props.activeTheme;
-        
-        //build out a base service context that can be cloned and passed to any 
+
+        //build out a base service context that can be cloned and passed to any
         //service objects
         this.context = {
             req: this.req,
             session: this.session,
             ls: this.ls,
             ts: this.ts,
-            activeTheme: this.activeTheme
+            site: this.site,
+            activeTheme: this.activeTheme,
+            onlyThisSite: true
         };
 
-        cb();
+        var siteService = new SiteService();
+        siteService.getByUid(this.site, function (err, siteInfo) {
+            if(pb.util.isError(err)) {
+                self.reqHandler.serve404();
+                return;
+            }
+            self.siteObj = siteInfo;
+            self.siteName = SiteService.isGlobal(siteInfo.uid) ? siteInfo.uid : siteInfo.displayName;
+            self.context.siteObj = self.siteObj;
+
+            self.ts.registerLocal('site_root', function(flag, cb) {
+                cb(null, SiteService.getHostWithProtocol(self.siteObj.hostname) || self.ts.siteRoot);
+            });
+            self.ts.registerLocal('site_name', function(flag, cb) {
+                cb(null, self.siteName);
+            });
+            cb();
+        });
```

Comment:
This is modified with the change of the DB lookup in base controller.  Let me know what you think of the implementation.

Ben Tidwell
Software Engineer III | Talent Network
CareerBuilder
5550-A Peachtree Pkwy, Norcross GA 30092

From: Brian <notifications@github.com<mailto:notifications@github.com>>
Reply-To: pencilblue/pencilblue <reply@reply.github.com<mailto:reply@reply.github.com>>
Date: Thursday, July 9, 2015 at 5:34 PM
To: pencilblue/pencilblue <pencilblue@noreply.github.com<mailto:pencilblue@noreply.github.com>>
Cc: Ben Tidwell <ben.tidwell@careerbuilder.com<mailto:ben.tidwell@careerbuilder.com>>
Subject: Re: [pencilblue] Multi Tenancy (#680)


In controllers/base_controller.js<https://github.com/pencilblue/pencilblue/pull/680#discussion_r34306509>:

> +            if(pb.util.isError(err)) {
> +                self.reqHandler.serve404();
> +                return;
> +            }
> +            self.siteObj = siteInfo;
> +            self.siteName = SiteService.isGlobal(siteInfo.uid) ? siteInfo.uid : siteInfo.displayName;
> +            self.context.siteObj = self.siteObj;
> +
> +            self.ts.registerLocal('site_root', function(flag, cb) {
> +                cb(null, SiteService.getHostWithProtocol(self.siteObj.hostname) || self.ts.siteRoot);
> +            });
> +            self.ts.registerLocal('site_name', function(flag, cb) {
> +                cb(null, self.siteName);
> +            });
> +            cb();
> +        });


If the site is truly required for each request then I don't think it should be the responsibility of the controller to retrieve the information. It should be handled as a step in the request pipeline (RequestHandler) before the controller is ever fired.

-
Reply to this email directly or view it on GitHub<https://github.com/pencilblue/pencilblue/pull/680/files#r34306509>.